src/main/scala/io/iohk/ethereum/network/discovery/PeerDiscoveryManager.scala

```diff
@@ -30,14 +30,19 @@ class PeerDiscoveryManager(
 
   var pingedNodes: Map[ByteString, PingInfo] = Map.empty
 
-  var nodesInfo: Map[ByteString, DiscoveryNodeInfo] = {
+  val startingNodes: Map[ByteString, DiscoveryNodeInfo] = {
     val knownNodesURIs =
       if (discoveryConfig.discoveryEnabled) knownNodesStorage.getKnownNodes()
       else Set.empty
-    val nodesInfo = knownNodesURIs.map(uri => DiscoveryNodeInfo.fromUri(uri)) ++ bootStrapNodesInfo
-    nodesInfo.map { nodeInfo => nodeInfo.node.id -> nodeInfo }.toMap
+    val startingNodesInfo = knownNodesURIs.map(uri => DiscoveryNodeInfo.fromUri(uri)) ++ bootStrapNodesInfo
+    val startingNodesInfoWithoutSelf = startingNodesInfo.filterNot {
+      _.node.id == ByteString(nodeStatusHolder.get().nodeId)
```

Comment:
shouldn't we filter `self` every time manager responds to `GetDiscoveredNodesInfo` message ?

Reply:
This depends on what discovery guarantees, given we are in the process of integrating our new discovery implementation I didn't want to get in the middle

(for example on the other project it does guarantee that it won't yourself as a discovered peer)

Reply:
thats a valid point. We can always fix it after integration with new discovery.